[bug 1353587] Make global nav accommodate text wrapping

craigcook opened the pull request:
## Description
The first iteration of the global nav had some hard-coded widths designed around the specific English links and didn't really handle links wrapping when they get lonk, which is almost guaranteed to happen once we start localizing those labels. This makes a few tweaks to the nav so when links wrap to a second line it doesn't look too horrible.

## Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1353587

## Testing
Since the navbar is only English right now you'll have to change some content locally to really test the wrapping. Try adding longer text in existing links as well as additional links, both in the main nav and subnav.

## Checklist
- [ ] Requires l10n changes.
- [ ] Related functional & integration tests passing.

alexgibson commented:
The other thing that occurs to me, is now that both navs wrap better, we *could* start using the regular desktop and tablet breakpoints for when we do thinks like hide the primary links, instead of using custom breakpoints for the nav. 

We may need to do some additional tweaks to accommodate this better (e.g. only applying 40px right padding to nav links for wide desktop), but it may make our breakpoint code easier to follow in the long run. Thoughts?

craigcook commented:
D'oh, of course I forgot the Sandstone version. And I'll work on switching to the standard breakpoints.

craigcook commented:
I've updated the Sandstone/Less nav and switched to our regular breakpoints (also removing the special nav breakpoints since they were no longer used anywhere).

alexgibson commented:
Looks like some of the mobile-first padding values are back to front on the .less version of the global nav? (if only we could just use the same, uniform vale across the board 🙄)

![padding](https://cloud.githubusercontent.com/assets/400117/26719330/fe70c7e0-477b-11e7-84b8-c780a74c12e3.png)

alexgibson commented:
@craigcook would you mind also pushing this to a demo? I can give things a check over for IE using browserstack, thanks.

alexgibson commented:
We should probably also update the sub-nav sticky break point to match these changes https://github.com/mozilla/bedrock/blob/master/media/js/hubs/sub-nav.js#L29

craigcook commented:
Switched padding around in Less (dangit), changed sub-nav.js breakpoint to 760px, and pushed to https://bedrock-demo-craigcook-nav.us-west.moz.works/

alexgibson commented:
Changes look good, although the right padding on links are still different between the scss and less versions (40px vs 20px)

alexgibson commented on media/css/pebbles/components/_global-nav.scss:
```diff
@@ -515,6 +531,7 @@ html.moz-nav-open .moz-global-nav-page-mask {
 
         .download-title {
             font-weight: bold;
+            white-space: nowrap;
```
I see this here in the scss version, but not the less version? Should they both be the same?

craigcook replied:
Hmm... well both should be the same, but now I'm not sure which way to go and maybe I should remove it from the SCSS. What do you think: let the button wrap or not?

alexgibson replied:
Well, for really long strings this could create overlapping text at some screen sizes, right? If that's a possibility, I'd say wrapping may be a bit more robust?

craigcook replied:
Removed!
